stream/sse/reader.go

```diff
@@ -0,0 +1,129 @@
+package sse
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+)
+
+func NewReader(ctx context.Context, rc io.ReadCloser) <-chan Event {
+	eventChan := make(chan Event)
+	go process(ctx, rc, eventChan)
+	return eventChan
+}
+
+func scanLines(data []byte, atEOF bool) (advance int, token []byte, err error) {
+	if atEOF && len(data) == 0 {
+		return 0, nil, nil
+	}
+
+	// Handle messages that use 'CRLF' or 'LF' line endings.
```

Comment:
Why do we need to handle CRLF endings?

Reply:
Well, if it's only used to read SSE messages from the writer in twofer, then we would not need to handle either CRLF or CR delimited messages. When I did the reader, I wanted it to follow the specification, so that it would be possible to use it to read SSE messages from other sources as well. This were when I intended to use SSE for the new v2 endpoints, before i discovered NDJSON. Since I already had everything in place for SSE, I kept it around in case it's simpler to use that in other languages, but if "caller" is using GO, it's simpler and much less code to use NDJSON, so I made NDJSON the default for the new v2 andpoints, but it's still possible to switch to SSE by setting an environment variable.

Reply:
Come to think about it now, maybe it would have been better to pass the intended delimiter as a parameter when creating the reader, that way the reader would not need the code that try to handle all possible line-endings.. Since when you are working with an SSE-stream, you would know what delimiter that particular stream uses as delimiter.